### PR TITLE
Add v5 upgrade guide

### PR DESCRIPTION
**Pull request by rmccue**
Still in progress, but pushing early.

### CODE REVIEW

File: other-docs/guides/upgrading/v5.md
Comment (joehoyle) [inline other-docs/guides/upgrading/v5.md:131]: So we recommend this? Seems like a pain to call out specifically.

**Review by joehoyle — APPROVED** [review]

File: other-docs/guides/upgrading/v5.md
Comment (rmccue) [reply other-docs/guides/upgrading/v5.md:131]: Might drop this section entirely as it seems to be backwards compatible, but had included as an "FYI our minimum of this dep you might also be using is being bumped"

File: other-docs/guides/upgrading/v5.md
Comment (roborourke) [inline other-docs/guides/upgrading/v5.md:55]: If we add core sitemaps in that can go here too

Comment (rmccue) [reply other-docs/guides/upgrading/v5.md:55]: Can handle in follow-up once merged?

File: other-docs/guides/upgrading/v5.md
Comment (rmccue) [reply other-docs/guides/upgrading/v5.md]: Did the default config for that change?

Comment (roborourke) [reply other-docs/guides/upgrading/v5.md]: No sorry I'm getting my wires crossed - they're actually separate. The `autosuggest` in config is really just for adding the client side search form dropdown now.

This does require reindexing though as it needs an updated mapping.

File: other-docs/guides/upgrading/v5.md
Comment (roborourke) [inline other-docs/guides/upgrading/v5.md:143]: GUessing we link those changelogs up once the stable is released?

File: other-docs/guides/upgrading/v5.md
Comment (roborourke) [inline other-docs/guides/upgrading/v5.md:91]:
```suggestion
This feature does require reindexing as it makes changes to the field mapping. [Reindexing can be performed at your discretion](docs://search/reindexing-content.md).
```

**Review by roborourke — APPROVED** [review]
Good to go unless you want to remove the bit about the AWS SDK

**Pull request merged**
